api/key/key.go

```diff
@@ -117,9 +117,48 @@ func Infer(k string, value interface{}) core.KeyValue {
 		return Float64(k, v)
 	case string:
 		return String(k, v)
+	case *bool:
```

Comment:
At the top of the function, consider adding a `if value == nil { return String(k, "nil") }`.